### PR TITLE
Joiners don't request data from other joiners during fast sync.

### PR DESCRIPTION
**Pull request by goral09**

### CODE REVIEW

File: node/src/components/chain_synchronizer/operations.rs [hunk 1]
```diff
@@ -203,7 +203,10 @@ where
     JoinerEvent: From<FetcherRequest<T>>,
 {
     loop {
-        let mut new_peer_list = ctx.effect_builder.get_fully_connected_peers().await;
+        let mut new_peer_list = ctx
+            .effect_builder
+            .get_fully_connected_validator_peers()
```
Comment (marc-casperlabs) [inline node/src/components/chain_synchronizer/operations.rs]: `non_joiner_peers`?

File: node/src/components/small_network/message.rs [hunk 2]
```diff
@@ -40,6 +40,9 @@ pub(crate) enum Message<P> {
         /// A self-signed certificate indicating validator status.
         #[serde(default)]
         consensus_certificate: Option<ConsensusCertificate>,
+        /// Joiner node flag.
+        #[serde(default)]
+        is_joiner: bool,
```
Comment (marc-casperlabs) [inline node/src/components/small_network/message.rs:45]: I wonder if `true` is the safer default here.

File: node/src/components/chain_synchronizer/operations.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/chain_synchronizer/operations.rs]: Done.

File: node/src/effect.rs [hunk 3]
```diff
@@ -675,6 +675,18 @@ impl<REv> EffectBuilder<REv> {
         .await
     }
 
+    /// Gets the current network peers in random order.
```
Comment (rafal-ch) [inline node/src/effect.rs]:
```suggestion
    /// Gets the current network non-joiner peers in random order.
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Pull request merged**